internal/consensus/msgs.go

```diff
@@ -269,11 +270,20 @@ func WALToProto(msg WALMessage) (*cmtcons.WALMessage, error) {
 		if err != nil {
 			return nil, err
 		}
+
+		var rtp *time.Time = nil
+		if !msg.ReceiveTime.IsZero() {
```

Comment:
I see you didn't need to modify the logic where the `ReceiveTime` field is populated/consumed. I'm assuming that's because that logic is reused from the PTBS code we have previously forward-ported. Two questions:
* Is this accurate?
* Are we sure this is addressing the problem we saw initially in tendermint/tendermint#8739 ?

Reply:
Yes, the second point is really important.

We need to make sure that a node doesn't equivocate after accepting/rejecting a `Proposal` during normal operation, same `Proposal` that will instead be untimely (most likely) or timely (e.g., ts in the future) upon recovery.

Reply:
This was tested by @sergio-mena using e2e